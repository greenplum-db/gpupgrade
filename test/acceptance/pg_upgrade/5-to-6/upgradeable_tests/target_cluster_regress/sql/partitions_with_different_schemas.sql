-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test partition tables whose children are in different schemas.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------

-- check data integrity after upgrade
SELECT * FROM public.different_schema_ptable ORDER BY 1, 2;
SELECT * FROM schema1.different_schema_ptable_1_prt_1 ORDER BY 1, 2;
SELECT * FROM schema2.different_schema_ptable_1_prt_2 ORDER BY 1, 2;
SELECT * FROM public.different_schema_ptable_1_prt_3 ORDER BY 1, 2;

-- check partition schemas
SELECT nsp.nspname, c.relname
FROM pg_class c
JOIN pg_namespace nsp ON nsp.oid = c.relnamespace
WHERE relname LIKE 'different_schema_ptable%'
ORDER BY relname;

-- test table insert
INSERT INTO public.different_schema_ptable VALUES (1, 2);
INSERT INTO public.different_schema_ptable VALUES (2, 2);
INSERT INTO public.different_schema_ptable VALUES (3, 2);

-- check data after insert
SELECT * FROM public.different_schema_ptable ORDER BY 1, 2;
SELECT * FROM schema1.different_schema_ptable_1_prt_1 ORDER BY 1, 2;
SELECT * FROM schema2.different_schema_ptable_1_prt_2 ORDER BY 1, 2;
SELECT * FROM public.different_schema_ptable_1_prt_3 ORDER BY 1, 2;



-- check data integrity after upgrade
SELECT * FROM public.multilevel_different_schema_ptable ORDER BY 1, 2, 3;
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_boys ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_1 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_2 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_3 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls ORDER BY 1, 2, 3;
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_girls_2_prt_1 ORDER BY 1, 2, 3;
SELECT * FROM schema2.multilevel_different_schema_ptable_1_prt_girls_2_prt_2 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls_2_prt_3 ORDER BY 1, 2, 3;

-- check partition schemas
SELECT nsp.nspname, c.relname
FROM pg_class c
JOIN pg_namespace nsp ON nsp.oid = c.relnamespace
WHERE relname LIKE 'multilevel_different_schema_ptable%'
ORDER BY relname;

-- test table insert
INSERT INTO public.multilevel_different_schema_ptable VALUES (7, date '2001-01-15', 'M');
INSERT INTO public.multilevel_different_schema_ptable VALUES (8, date '2002-02-15', 'M');
INSERT INTO public.multilevel_different_schema_ptable VALUES (9, date '2003-03-15', 'M');
INSERT INTO public.multilevel_different_schema_ptable VALUES (10, date '2001-01-15', 'F');
INSERT INTO public.multilevel_different_schema_ptable VALUES (11, date '2002-02-15', 'F');
INSERT INTO public.multilevel_different_schema_ptable VALUES (12, date '2003-03-15', 'F');

-- check data after insert
SELECT * FROM public.multilevel_different_schema_ptable ORDER BY 1, 2, 3;
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_boys ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_1 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_2 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_3 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls ORDER BY 1, 2, 3;
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_girls_2_prt_1 ORDER BY 1, 2, 3;
SELECT * FROM schema2.multilevel_different_schema_ptable_1_prt_girls_2_prt_2 ORDER BY 1, 2, 3;
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls_2_prt_3 ORDER BY 1, 2, 3;
