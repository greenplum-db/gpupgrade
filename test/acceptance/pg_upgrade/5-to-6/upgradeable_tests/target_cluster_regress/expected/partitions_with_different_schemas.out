-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test partition tables whose children are in different schemas.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------

-- check data integrity after upgrade
SELECT * FROM public.different_schema_ptable ORDER BY 1, 2;
 a | b 
---+---
 1 | 1 
 2 | 1 
 3 | 1 
(3 rows)
SELECT * FROM schema1.different_schema_ptable_1_prt_1 ORDER BY 1, 2;
 a | b 
---+---
 1 | 1 
(1 row)
SELECT * FROM schema2.different_schema_ptable_1_prt_2 ORDER BY 1, 2;
 a | b 
---+---
 2 | 1 
(1 row)
SELECT * FROM public.different_schema_ptable_1_prt_3 ORDER BY 1, 2;
 a | b 
---+---
 3 | 1 
(1 row)

-- check partition schemas
SELECT nsp.nspname, c.relname FROM pg_class c JOIN pg_namespace nsp ON nsp.oid = c.relnamespace WHERE relname LIKE 'different_schema_ptable%' ORDER BY relname;
 nspname | relname                         
---------+---------------------------------
 public  | different_schema_ptable         
 schema1 | different_schema_ptable_1_prt_1 
 schema2 | different_schema_ptable_1_prt_2 
 public  | different_schema_ptable_1_prt_3 
(4 rows)

-- test table insert
INSERT INTO public.different_schema_ptable VALUES (1, 2);
INSERT 1
INSERT INTO public.different_schema_ptable VALUES (2, 2);
INSERT 1
INSERT INTO public.different_schema_ptable VALUES (3, 2);
INSERT 1

-- check data after insert
SELECT * FROM public.different_schema_ptable ORDER BY 1, 2;
 a | b 
---+---
 1 | 1 
 1 | 2 
 2 | 1 
 2 | 2 
 3 | 1 
 3 | 2 
(6 rows)
SELECT * FROM schema1.different_schema_ptable_1_prt_1 ORDER BY 1, 2;
 a | b 
---+---
 1 | 1 
 1 | 2 
(2 rows)
SELECT * FROM schema2.different_schema_ptable_1_prt_2 ORDER BY 1, 2;
 a | b 
---+---
 2 | 1 
 2 | 2 
(2 rows)
SELECT * FROM public.different_schema_ptable_1_prt_3 ORDER BY 1, 2;
 a | b 
---+---
 3 | 1 
 3 | 2 
(2 rows)



-- check data integrity after upgrade
SELECT * FROM public.multilevel_different_schema_ptable ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 1  | 01-15-2001 | M      
 2  | 02-15-2002 | M      
 3  | 03-15-2003 | M      
 4  | 01-15-2001 | F      
 5  | 02-15-2002 | F      
 6  | 03-15-2003 | F      
(6 rows)
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_boys ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 1  | 01-15-2001 | M      
 2  | 02-15-2002 | M      
 3  | 03-15-2003 | M      
(3 rows)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_1 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 1  | 01-15-2001 | M      
(1 row)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_2 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 2  | 02-15-2002 | M      
(1 row)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_3 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 3  | 03-15-2003 | M      
(1 row)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 4  | 01-15-2001 | F      
 5  | 02-15-2002 | F      
 6  | 03-15-2003 | F      
(3 rows)
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_girls_2_prt_1 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 4  | 01-15-2001 | F      
(1 row)
SELECT * FROM schema2.multilevel_different_schema_ptable_1_prt_girls_2_prt_2 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 5  | 02-15-2002 | F      
(1 row)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls_2_prt_3 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 6  | 03-15-2003 | F      
(1 row)

-- check partition schemas
SELECT nsp.nspname, c.relname FROM pg_class c JOIN pg_namespace nsp ON nsp.oid = c.relnamespace WHERE relname LIKE 'multilevel_different_schema_ptable%' ORDER BY relname;
 nspname | relname                                                
---------+--------------------------------------------------------
 public  | multilevel_different_schema_ptable                     
 schema1 | multilevel_different_schema_ptable_1_prt_boys          
 public  | multilevel_different_schema_ptable_1_prt_boys_2_prt_1  
 public  | multilevel_different_schema_ptable_1_prt_boys_2_prt_2  
 public  | multilevel_different_schema_ptable_1_prt_boys_2_prt_3  
 public  | multilevel_different_schema_ptable_1_prt_girls         
 schema1 | multilevel_different_schema_ptable_1_prt_girls_2_prt_1 
 schema2 | multilevel_different_schema_ptable_1_prt_girls_2_prt_2 
 public  | multilevel_different_schema_ptable_1_prt_girls_2_prt_3 
(9 rows)

-- test table insert
INSERT INTO public.multilevel_different_schema_ptable VALUES (7, date '2001-01-15', 'M');
INSERT 1
INSERT INTO public.multilevel_different_schema_ptable VALUES (8, date '2002-02-15', 'M');
INSERT 1
INSERT INTO public.multilevel_different_schema_ptable VALUES (9, date '2003-03-15', 'M');
INSERT 1
INSERT INTO public.multilevel_different_schema_ptable VALUES (10, date '2001-01-15', 'F');
INSERT 1
INSERT INTO public.multilevel_different_schema_ptable VALUES (11, date '2002-02-15', 'F');
INSERT 1
INSERT INTO public.multilevel_different_schema_ptable VALUES (12, date '2003-03-15', 'F');
INSERT 1

-- check data after insert
SELECT * FROM public.multilevel_different_schema_ptable ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 1  | 01-15-2001 | M      
 2  | 02-15-2002 | M      
 3  | 03-15-2003 | M      
 4  | 01-15-2001 | F      
 5  | 02-15-2002 | F      
 6  | 03-15-2003 | F      
 7  | 01-15-2001 | M      
 8  | 02-15-2002 | M      
 9  | 03-15-2003 | M      
 10 | 01-15-2001 | F      
 11 | 02-15-2002 | F      
 12 | 03-15-2003 | F      
(12 rows)
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_boys ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 1  | 01-15-2001 | M      
 2  | 02-15-2002 | M      
 3  | 03-15-2003 | M      
 7  | 01-15-2001 | M      
 8  | 02-15-2002 | M      
 9  | 03-15-2003 | M      
(6 rows)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_1 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 1  | 01-15-2001 | M      
 7  | 01-15-2001 | M      
(2 rows)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_2 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 2  | 02-15-2002 | M      
 8  | 02-15-2002 | M      
(2 rows)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_boys_2_prt_3 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 3  | 03-15-2003 | M      
 9  | 03-15-2003 | M      
(2 rows)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 4  | 01-15-2001 | F      
 5  | 02-15-2002 | F      
 6  | 03-15-2003 | F      
 10 | 01-15-2001 | F      
 11 | 02-15-2002 | F      
 12 | 03-15-2003 | F      
(6 rows)
SELECT * FROM schema1.multilevel_different_schema_ptable_1_prt_girls_2_prt_1 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 4  | 01-15-2001 | F      
 10 | 01-15-2001 | F      
(2 rows)
SELECT * FROM schema2.multilevel_different_schema_ptable_1_prt_girls_2_prt_2 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 5  | 02-15-2002 | F      
 11 | 02-15-2002 | F      
(2 rows)
SELECT * FROM public.multilevel_different_schema_ptable_1_prt_girls_2_prt_3 ORDER BY 1, 2, 3;
 id | year       | gender 
----+------------+--------
 6  | 03-15-2003 | F      
 12 | 03-15-2003 | F      
(2 rows)
